astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1025 @@
+#define NO_IMPORT_ARRAY
+#include <math.h>
+#include <float.h>
+
+#include "astropy_wcs/wcslib_celprm_wrap.h"
+#include "astropy_wcs/wcslib_prjprm_wrap.h"
+
+#include <wcs.h>
+#include <wcsprintf.h>
+#include <prj.h>
+#include <numpy/npy_math.h>
+#include "astropy_wcs/docstrings.h"
+
+
+PyObject** prj_errexc[5];
+
+
+static int is_dbl_equal(double x1, double x2)
+{
+    double ax1 = fabs(x1);
+    double ax2 = fabs(x2);
+    double minx = (ax1 < ax2) ? ax1 : ax2;
+    return (fabs(x1 - x2) <= 2.0 * DBL_EPSILON * minx);
```

Comment:
In the very very unfortunate case when this condition (in this PR) is not satisfied but it would satisfy the second in the linked code, the worst thing that is going to happen is that `set()` will be called. But, OK, let's add a second check for those cases when one of the compared values is 0 or almost 0.